### DISCUSSION TITLE
WordPress Plugin or how to stop stream after time

### DISCUSSION THREAD

**Discussion by AxelHamburch** [bluenviron/mediamtx#2651] [Questions]
### Question

First of all, thank you very much for this great software. It works wonderfully and is really a universal tool. Absolutely great!

I have set up a server that outputs the stream as WebRTC in the form http://localhost:8889/mystream. If I call up the link with the customized IP of my server in a browser, I can also see my stream after a short time. 

Now I would like to integrate this into my WordPress website. It doesn't work as a direct link and it doesn't work as a video either. Unfortunately, I have not found a suitable plugin that can do WebRTC. I can only call up a new page and then it also streams. 

However, firstly I would like to display the stream directly on my page and secondly I would like to start the stream from the page and at best have it stop automatically after a certain time. But at least the stream should be automatically terminated/stopped when a page change occurs. 

The problem is that otherwise the stream continues to run forever until you actively close the website. Especially on mobile devices, you have several pages in the background because you open new ones or simply switch away from the browser view. If the stream is then not stopped automatically, it will continue to run without you noticing. 

Is there a setting or a plugin for WordPress that can handle this? Only in the direct view should the stream run or because of mine also switch off after time and you have to actively switch on again. Is there a possibility or a work around?

**Comment by aler9** [comment]
Hello, there are several ways to integrate a stream from the server into Wordpress or any other web page. The simpler one consists in using an iframe, as reported in the README:

```html
<iframe src="http://mediamtx-ip:8889/mystream/publish" scrolling="no"></iframe>
```

You can dynamically start and stop the stream by creating and removing the iframe through JavaScript, for instance, in order to stop the stream after 10 seconds:

```html
<iframe src="http://mediamtx-ip:8889/mystream/publish" scrolling="no" id="mystream"></iframe>
<script>
setTimeout(() => {
    document.getElementById("mystream").src = "";
}, 10000);
</script>
```

This has the advantage that all the WebRTC management, including automatic reconnection, is automatically handled for you by the server.

If you instead want to use the `<video>` element directly, you can copy and paste the [read_index.html page](https://github.com/bluenviron/mediamtx/blob/main/internal/servers/webrtc/read_index.html) from the server and change it in a way that suits your needs.

[a comment by an automated account is not shown]
